### PR TITLE
Add GooseAI, CerebriumAI, Petals, ForefrontAI

### PR DESCRIPTION
**Pull request by conceptofmind**
Add GooseAI, CerebriumAI, Petals, ForefrontAI

### CODE REVIEW

**Comment by batmanscode** [comment]
Nice! 😃

This will close #815 and #875

File: langchain/llms/__init__.py
Comment (batmanscode) [inline langchain/llms/__init__.py:42]: This looks like the cause of the failing lint test. `"gooseai": "GooseAI"` should be `"gooseai": GooseAI`

Error:
https://github.com/hwchase17/langchain/actions/runs/4146938778/jobs/7173220060#step:6:20

image

Comment (conceptofmind) [reply langchain/llms/__init__.py:42]: Ok.Thank you for bringing attention to this. I am unfamiliar with Lint. I will resolve this.

Comment (conceptofmind) [reply langchain/llms/__init__.py]: @batmanscode It looks like everything is passing now. Thank you again.

File: docs/modules/llms/integrations/petals_example.ipynb
Comment (batmanscode) [inline docs/modules/llms/integrations/petals_example.ipynb:8]: What do you think about including a note that it'll download ~8GB for this? I think this would require the user to have a GPU as well but not certain.

This info is from the Colab getting started tutorial linked on their github page:
https://colab.research.google.com/drive/1Ervk6HPNS6AYVr3xVdQnY5a-TjjmLCdQ?usp=sharing
image

File: langchain/llms/gooseai.py
Comment (batmanscode) [inline langchain/llms/gooseai.py:61]: I think it'll be great to add the goose specific parameters (https://goose.ai/docs/api/completions) they've implemented, like `typical_p`. This, in addition to their low cost, are a big advantage to using their service 😃 

image

Or maybe just make a note that's easily visible to check goose docs and let them be kwargs. What do you think?

**Review by batmanscode — COMMENTED** [review]
Looks good, thanks for adding these 😃!

Your PR is good as it is. I've suggested some minor changes to include additional info. They don't need to be changed but bringing to your attention in case you feel the same way

File: langchain/llms/__init__.py
Comment (batmanscode) [reply langchain/llms/__init__.py]: Nice, glad it worked!

**Comment by batmanscode** [comment]
Also, you should probably request a review from @hwchase17 as he has write access

File: docs/modules/llms/integrations/petals_example.ipynb
Comment (conceptofmind) [reply docs/modules/llms/integrations/petals_example.ipynb:8]: Petals does not require a GPU for inference. I can look into adding functionality for automatically checking if a GPU is available for inference.

File: langchain/llms/gooseai.py
Comment (conceptofmind) [reply langchain/llms/gooseai.py:61]: I think `model_kwargs` would cover this?

**Comment by conceptofmind** [comment]
> Also, you should probably request a review from @hwchase17 as he has write access

Thank you for the additional input. I think once the basic functionality is merged we could do an extensive review of all the documentation? You could also do a separate PR so that you receive credit for your work. I had messaged Chase on discord previously about this PR.

**Review by hwchase17 — APPROVED** [review]
thank you both for your pr + comments!

agree more thorough review of pr is necc

**Comment by batmanscode** [comment]
> > Also, you should probably request a review from @hwchase17 as he has write access
> 
> Thank you for the additional input. I think once the basic functionality is merged we could do an extensive review of all the documentation? You could also do a separate PR so that you receive credit for your work. I had messaged Chase on discord previously about this PR.

Sure can do 😃

And thanks again for this PR been wanting to use goose.ai for a while now! 😆 also didn't even know Cerebrium existed

File: langchain/llms/gooseai.py
Comment (batmanscode) [reply langchain/llms/gooseai.py:61]: Yes sir 😃

**Pull request merged**
